### PR TITLE
Dashboards: Show panel load failure error state

### LINKED ISSUE #73918: Exceptions in panel migration handlers are silently ignored
###  What happens

If an error is thrown in `PanelModel.pluginLoaded` when initialising a panel, such as in the panel migration handler, the panel is just not rendered and no error is shown.

You can reproduce this just by adding a `throw new Error()` in `PanelModel.pluginLoaded`.

This dashboard should have 5 panels, but nothing is shown when the migration handler fails for all of them 

image

Confusingly, if you the panel off screen and back again, it'll show. This seems weird though because now the panel is shown even though it was not completely initialised - it'll be in a weird partial state.

### What's expected

At a bare minimum, I would expect an error toast is shown. But ideally, if the panel/plugin/migration handler fails, then the panel should be rendered in an error state.

### PR DESCRIPTION
**Pull request by joshhunt**
Initial stab at showing errors for when panels fail to initialise

Sharing draft now to see if this is the right way to guard for these errors. Maybe there might be a better place higher in the call stack? Unsure.

image

fixes https://github.com/grafana/grafana/issues/73918

### CODE REVIEW

File: public/app/features/panel/components/PanelPluginError.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ class PanelPluginError extends PureComponent<Props> {
 
     return (
       <div style={style}>
-        <Alert severity={AppNotificationSeverity.Error} {...this.props} />
+        <Alert severity={AppNotificationSeverity.Error} children={this.props.text} {...this.props} />
```
Comment (polibb) [inline public/app/features/panel/components/PanelPluginError.tsx:29]: This is temporary, right 😄

**Comment by polibb** [comment]
Hi @joshhunt, thank you for taking initial steps towards fixing this overlooked error handling. Let me just confirm I got the scenario correctly, correct me if needed:
panel to be initialised -> plugin to be loaded -> plugin is different version -> migrate to current version -> API call to migrate it -> API response is error -> error in loading plugin -> error in initialising panel

[This is where the error is thrown](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/PanelModel.ts#L452-L458), right?

At first I thought it's alright to handle all errors produced by actions in `initPanelState()` in `initPanelState()` itself, but it's hard to follow through why it's needed - that actually `loadedPlugin()` is the one erroring. 
I think it's better to do something specific to the "sore spot", [like here](https://github.com/grafana/grafana/blob/main/public/app/features/panel/state/actions.ts#L23-L30). The `try catch` surrounds what is necessary only.

File: public/app/features/panel/components/PanelPluginError.tsx (hunk 1 above)
Comment (joshhunt) [reply public/app/features/panel/components/PanelPluginError.tsx:29]: I don't see why it would be? These error components are passed `text` which isn't used, because the props are incorrectly spread.

Comment (polibb) [reply public/app/features/panel/components/PanelPluginError.tsx:29]: Yes, I'm glad you fixed that - that we're not using `text` is something I noticed only after seeing your PR. I'm just talking about the `children` prop, which eslint is complaining about when you run the FE: 
>ERROR in [eslint] 
/Users/polibon/Documents/code/grafana/grafana/public/app/features/panel/components/PanelPluginError.tsx
  29:57  error  Do not pass children as props. Instead, nest children between the opening and closing tags  react/no-children-prop
